app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Admin::UsersController < Admin::BaseController
+  before_action :set_user, only: [:enable_or_disable_user, :promote_to_manager_or_demote_to_user]
+  
+  def index
+    @users = User.all_users_except_admin
+  end
+  
+  def enable_or_disable_user
+    if @user.disabled?
```

Comment:
I think some refactoring needed here. I think we should do the following here:

- Shift enable/disabled logic to model.
- Rename method from `enable_or_disable_user ` to `update_role`.
- There must be only one `respond_to` block.

Reply:
Implemented as suggested.